rosidl_generator_c/rosidl_generator_c/__init__.py

```diff
@@ -34,7 +34,20 @@ def generate_c(generator_arguments_file):
         'idl__struct.h.em': '%s__struct.h',
         'idl__type_support.h.em': '%s__type_support.h',
     }
-    generate_files(generator_arguments_file, mapping)
+    generate_files(
+        generator_arguments_file, mapping,
+        post_process_callback=prefix_with_bom_if_necessary)
+
+
+def prefix_with_bom_if_necessary(content):
+    try:
+        content.encode('ASCII')
+    except UnicodeError:
+        prefix = '\ufeff' + \
```

Comment:
[2.13 Page 58 (pdf page 52)](https://www.unicode.org/versions/Unicode5.0.0/ch02.pdf) says the utf-8 BOM is `0xEF, 0xBB, 0xBF`. This looks like the UTF-16 BOM.

Reply:
I just replicated what we already use in the only existing file with a BOM: https://github.com/ros2/rcl_interfaces/blob/75e94e1d10399fda56ccb7c635aef730e3d7def0/test_msgs/include/test_msgs/message_fixtures.hpp#L1

`hexdump -n 3 -C <filename>` returns `ef bb bf`

Reply:
Ah oops, I get it. `\ufeff` is the code point, and `ef bb bf` are the encoded bytes. 

```python
>>> u"\ufeff".encode('utf-8')
'\xef\xbb\xbf'
```